product_docs/docs/pgd/5/routing/proxy.mdx

```diff
@@ -20,20 +20,34 @@ PGD Proxy is a TCP layer 4 proxy.
 Upon starting, PGD Proxy connects to one of the endpoints given in the local config file. It fetches:
 
 -  DB connection information for all nodes
--  Proxy options like listen address, listen port 
+-  Proxy options like listen address, listen port
 -  Routing details like current write leader
 
 Endpoints given in the config file are used only at startup. After that, actual endpoints are taken from the PGD catalog's `route_dsn` field in `bdr.node_routing_config_summary`.
 
 
-PGD manages write leader election. PGD Proxy interacts with PGD to get write leader change events notifications on Postgres notify/listen channels and routes client traffic to the current write leader. PGD Proxy disconnects all client connections on write leader change. Write leader election is a Raft-backed activity and is subject to Raft leader availability.
+PGD manages write leader election. PGD Proxy interacts with PGD to get write leader change events notifications on Postgres notify/listen channels and routes client traffic to the current write leader. PGD Proxy disconnects all existing client connections on write leader change or when write leader is unavailable. Write leader election is a Raft-backed activity and is subject to Raft leader availability. Proxy will close the new client connections if write leader is unavailable.
 
 PGD Proxy responds to write leader change events that can be categorized into two modes of operation: *failover* and *switchover*.
 
 Automatic transfer of write leadership from the current write leader node to a new node in the event of Postgres or operating system crash is called failover. PGD elects a new write leader when the current write leader goes down or becomes unresponsive. Once the new write leader is elected by PGD, proxy closes existing client connections to the old write leader and redirects new client connections to the newly elected write leader.
 
 User-controlled, manual transfer of write leadership from the current write leader to a new target leader is called switchover. Switchover is triggered through the [PGD CLI switchover](../cli/command_ref/pgd_switchover) command. The command is submitted to PGD, which attempts to elect the given target node as the new write leader. Similar to failover, PGD Proxy closes existing client connections and redirects new client connections to the newly elected write leader. This is useful during server maintenance, for example, if the current write leader node needs to be stopped for maintenance like a server update or OS patch update.
 
+### Consensus Grace Period
+
+PGD Proxy provides the [consensus_grace_period](index.mdx) proxy option that can be used to configure the routing behaviour upon loss of a Raft leader. Proxy will continue to route to the current write leader (if it is available) for this duration. If the new Raft leader is not elected during this period, the proxy will stop routing. If set to `0s` proxy will stop routing immediately.
```

Comment:
I suggest you lose the link here for now as it only takes you to the top of a page where there's an entry and the entry has less detail than the sentence you are currently in.